dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -610,14 +606,18 @@ public Map<String, Object> grantProjectByCode(final User loginUser, final int us
         }
 
         // 4. maintain the relationship between project and user
-        final Date today = new Date();
-        ProjectUser projectUser = new ProjectUser();
-        projectUser.setUserId(userId);
-        projectUser.setProjectId(project.getId());
-        projectUser.setPerm(7);
-        projectUser.setCreateTime(today);
-        projectUser.setUpdateTime(today);
-        this.projectUserMapper.insert(projectUser);
+        // check if already existed
+        ProjectUser pu = projectUserMapper.queryProjectRelation(project.getId(), userId);
```

Comment:
Same as ```projectId```.

Reply:
but this `getId` is object `project`'s function

Reply:
thanks.but id is the property of project and should not change

Reply:
@leiwingqueen Hi,variable pu did not change to projectUser.

Reply:
got it.